taca/utils/bioinfo_tab.py

```diff
@@ -24,54 +25,82 @@ def __init__(self, value=None):
 
 def collect_runs():
     """Update command."""
-    found_runs = []
+    found_runs = {"illumina": [], "element": []}
     # Pattern explained:
     # 6-8Digits_(maybe ST-)AnythingLetterornumberNumber_Number_AorBLetterornumberordash
-    rundir_re = re.compile("\d{6,8}_[ST-]*\w+\d+_\d+_[AB]?[A-Z0-9\-]+")
-    for data_dir in CONFIG["bioinfo_tab"]["data_dirs"]:
-        if os.path.exists(data_dir):
-            potential_run_dirs = glob.glob(os.path.join(data_dir, "*"))
-            for run_dir in potential_run_dirs:
-                if rundir_re.match(
-                    os.path.basename(os.path.abspath(run_dir))
-                ) and os.path.isdir(run_dir):
-                    found_runs.append(os.path.basename(run_dir))
-                    logger.info(f"Working on {run_dir}")
-                    update_statusdb(run_dir)
-        nosync_data_dir = os.path.join(data_dir, "nosync")
-        potential_nosync_run_dirs = glob.glob(os.path.join(nosync_data_dir, "*"))
-        for run_dir in potential_nosync_run_dirs:
-            if rundir_re.match(
-                os.path.basename(os.path.abspath(run_dir))
-            ) and os.path.isdir(run_dir):
-                update_statusdb(run_dir)
-
-
-def update_statusdb(run_dir):
+    illumina_rundir_re = re.compile("\d{6,8}_[ST-]*\w+\d+_\d+_[AB]?[A-Z0-9\-]+")
+    for inst_brand in CONFIG["bioinfo_tab"]["data_dirs"]:
+        for data_dir in CONFIG["bioinfo_tab"]["data_dirs"][inst_brand]:
+            if os.path.exists(data_dir):
+                potential_run_dirs = glob.glob(os.path.join(data_dir, "*"))
+                for run_dir in potential_run_dirs:
+                    if os.path.isdir(run_dir):
+                        if inst_brand == "illumina" and illumina_rundir_re.match(
+                            os.path.basename(os.path.abspath(run_dir))
+                        ):
+                            found_runs["illumina"].append(os.path.basename(run_dir))
+                            logger.info(f"Working on {run_dir}")
+                            update_statusdb(run_dir, inst_brand)
+                        elif inst_brand == "element":
+                            logger.info(f"Working on {run_dir}")
```

Comment:
Don't we want to have some kind of regular expression to check for a rundir here as well? I think there is a risk there will be other directories inside the data_dir? nosync for example?

Reply:
Added checks for `nosync` and `archived` dirs in the data directories. Also added an try except below to catch `FileNotFoundError` in case the run is not ready yet and it is not a run dir.